lib/compiler.js

```diff
@@ -598,3 +604,69 @@ function resolveAppScriptPath(rootDir, relativePath, resolveOptions) {
   var fixedFile = fixFileExtension(resolvedPath, files, false);
   return (fixedFile === undefined ? resolvedPath : fixedFile);
 }
+
+function buildAllMixinInstructions(appRootDir, mixinDirs, options) {
+  var extensions = _.without(_.keys(require.extensions),
+    _.keys(getExcludedExtensions()));
+  var files = options.mixins || [];
+
+  mixinDirs.forEach(function(dir) {
+    dir = tryResolveAppPath(appRootDir, dir);
```

Comment:
I guess the comment is not clear enough then.

For component paths (i.e. when resolving keys in `component-config.json`), the backwards-compatible behaviour is to always resolve "foo" as "node_modules/foo" and never as "./foo".

On the other hand, for some other places like model sources, the backwards-compatible behaviour is to resolve "foo" as "./foo" or "node_modules/foo".

The strict option controls this behaviour. Strict:true = search `node_modules` only. Strict:false = try a relative path too.

Now my argument is that all new code should use strict:true mode and therefore it's better to change `tryResolveAppPath` to default to strict:true.

I have made the cleanup myself, could you please review https://github.com/strongloop/loopback-boot/pull/127?